### PR TITLE
Travis: add back cache, and reduce verbose on git commands

### PR DESCRIPTION
**Pull request by khancyr**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,7 +30,7 @@ addons:
       - llvm-7
 
 cache:
-  ccache: false
+  ccache: true
```
Comment (OXINARF) [inline .travis.yml]: No, this is worse than without saving ccache (the savings in compile time aren't enough for the time wasted saving and restoring the cache), which is why it was turned off.

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,10 @@
 language: cpp
 sudo: false
 
+git:
+  depth: 10
+  quiet: true
```
Comment (OXINARF) [inline .travis.yml]: I'll have to re-check how depth works, if it is passed to submodules this isn't going to work well with branches (submodule commits won't be reached). I think I've tried it in the past, but I don't really remember it.

As for quiet, I'd also say no, that hides important information for when problems occur.

**Comment by khancyr** [comment]
hum that is weird that the caching isn't worth it. The main purpose of this PR was to speed up the build phase and prevent unnecessary rebuild between push. 
It is more a WIP as I want to check the timing.

**Comment by magicrub** [comment]
@khancyr have you done any benchmark time tests to see how long the compile takes? I suppose it wouldn't hurt to ad a "time" call to the bigger script calls so we can start seeing CI test times for everything

**Comment by khancyr** [comment]
closing for now ... caching doesn't seems to work ...
I still think it is worth it as it will reduce build phase by 2m50s for each build. Totally worth it from waiting and ecological perspective

**Pull request closed without merging**